src/oss/langgraph/workflows-agents.mdx

```diff
@@ -0,0 +1,2221 @@
+---
+title: Workflows and agents
+---
+
+This guide reviews common workflow and agent patterns. Workflows have predetermined code paths and are designed to operate in a certain order. Agents are more dynamic and define their own processes and tool usage.
+
+![Agent Workflow](/oss/images/agent_workflow.png)
+
+LangGraph offers several benefits when building agents and workflows, including persistence, streaming, and support for debugging as well as deployment.
+
+## Setup
+
+You can use [any chat model](/oss/integrations/chat) that supports structured outputs and tool calling to build workflows and agents. To get started, install dependencies and then set your API keys like the Anthropic example below.
+
+:::python
+<Accordion title="Install dependencies">
+```bash
+pip install langchain_core langchain-anthropic langgraph
+```
+</Accordion>
+
+With dependencies installed, you can initialize the LLM.
+
+```python
+import os
+import getpass
+
+from langchain_anthropic import ChatAnthropic
+
+def _set_env(var: str):
+    if not os.environ.get(var):
+        os.environ[var] = getpass.getpass(f"{var}: ")
+
+
+_set_env("ANTHROPIC_API_KEY")
+
+llm = ChatAnthropic(model="claude-3-5-sonnet-latest")
+```
+:::
+
+:::js
+<Accordion title="Install dependencies">
+```bash
+npm install @langchain/core @langchain/anthropic @langchain/langgraph
+```
+</Accordion>
+
+With dependencies installed, you can initialize the LLM.
+
+```typescript
+import { ChatAnthropic } from "@langchain/anthropic";
+
+process.env.ANTHROPIC_API_KEY = "<your_anthropic_key>";
+
+const llm = new ChatAnthropic({
+  model: "claude-3-5-sonnet-latest",
+});
+```
+:::
+
+## LLMs and augmentations
+
+Workflows and agentic systems are based on LLMs and the various augmentations you add to them. [Tool calling](/oss/langchain-tools), [structured outputs](/oss/langchain-structured-output), and [short term memory](/oss/langchain-short-term-memory) are a few options for tailoring LLMs to your needs.
+
+![LLM augmentations](/oss/images/augmented_llm.png)
+
+### Code examples
+
+:::python
```

Comment:
Should we need to add code samples for the functional API too here since we show both Graph API + Functional API in following sections?